firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatController.java

```diff
@@ -95,19 +96,20 @@
               HeartBeatResult result = allHeartBeats.get(i);
               JSONObject obj = new JSONObject();
               obj.put("agent", result.getUserAgent());
-              obj.put("date", result.getUsedDates());
+              obj.put("dates", new JSONArray(result.getUsedDates()));
               array.put(obj);
             }
             JSONObject output = new JSONObject();
             output.put("heartbeats", array);
             output.put("version", "2");
             ByteArrayOutputStream out = new ByteArrayOutputStream();
-            GZIPOutputStream gzip = new GZIPOutputStream(out);
-            gzip.write(output.toString().getBytes());
+            Base64OutputStream b64os =
+                new Base64OutputStream(out, Base64.URL_SAFE | Base64.NO_PADDING | Base64.NO_WRAP);
+            GZIPOutputStream gzip = new GZIPOutputStream(b64os);
+            gzip.write(output.toString().getBytes("UTF-8"));
```

Comment:
This whole block can be more readable and conventional by using try with resources, something like:

```java
try(ByteArrayOutputStream out = new ByteArrayOutputStream();
    Base64OutputStream(out, Base64.URL_SAFE | Base64.NO_PADDING | Base64.NO_WRAP);
    GZIPOutputStream gzip = new GZIPOutputStream(b64os)) {
  gzip.write(output.toString().getBytes("UTF-8"));
  return out.toString("UTF-8");
}
```